### ISSUE TITLE
MIDI Input doesn't work on Windows 10 - BROKEN

### ISSUE THREAD

**Issue by HeyItsJustMe** [frescobaldi/frescobaldi#853]
Sorry but MIDI input is not working on Windows 10 64bit. I've tried several MIDI keyboards, none work. But I can confirm MIDI is being transmitted correctly from these devices, but Frescobaldi will not do MIDI input. It just does nothing. Start and stop capture = completely broken on Win10.  MIDI-OX shows perfectly fine MIDI data transfer FYI.

Can this be fixed please? It's such a hassle now. Thank you.

Note: I'm happy to report that the MIDI devices are working with MIDI INPUT in Frescobaldi on Windows 7 64. This is ONLY a problem with Windows 10 and 8.1 and Frescobaldi MIDI INPUT. Really hope you can fix this soon.

**Comment by drydh** [comment]
MIDI input doesn't work for me either (on Mac OS 10.10). MIDI output is fine. This is with Frescobaldi 2.19.0. I'll investigate this further when I've been able to get pyportmedia/pypm/pygame running under Python3 so I can debug (or I'll get the Python 2.x version running).

Migrating to mido (#579) or pyrtmidi would perhaps fix these issues but could involve quite some work.

**Comment by HeyItsJustMe** [comment]
But it works perfectly fine in Windows 7. What changed with Windows 10 that MIDI input is so terribly unusable? It is great in Win7.

Such a headache. I love Frescobaldi but this is a tough situation.

**Comment by drydh** [comment]
I found the bug causing my problem: Frescobaldi has **never** honored the choice of the midi input device in the preferences due to a confusion between the prefixes 'midi.midi' and 'midi'! It therefore always uses the default input device. To check if this is also causing your problem, find the preferences for Frescobaldi (in the registry on Windows?) and change the key midi.midi.input_port to midi.input_port and see if this fixes your problem.

**Comment by HeyItsJustMe** [comment]
Would that have been really so different on Windows 7 where it only would cause a problem on 8.1 and Win10? Odd. I will try your suggestion and post my results here. Thank you for the tip!

**Comment by drydh** [comment]
I don't have a Windows machine but looking at other issues, the key in the registry is probably (at least under Windows 8.1) called:

HHKEY_CURRENT_USER\software\Frescobaldi\frescobaldi\midi\midi\input_port

and should be changed to

HHKEY_CURRENT_USER\software\Frescobaldi\frescobaldi\midi\input_port

**Comment by drydh** [comment]
For further information on the location of Frescobaldi preferences, see [QT Documentation](http://doc.qt.io/qt-5/qsettings.html#platform-specific-notes).

**Comment by unmostro** [comment]
drydh, thanks a lot - your method works!
I've opened the registry (on Windows 10), copied input_port item from HKEY_CURRENT_USER\SOFTWARE\frescobaldi\frescobaldi\midi\midi
to 
HKEY_CURRENT_USER\SOFTWARE\frescobaldi\frescobaldi\midi,
set it to correct value and voila - at last I can enjoy midi input right in Frescobaldi.

**Comment by HeyItsJustMe** [comment]
I hope this can be fixed in the next Frescobaldi update :) This registry trick was so genius thank you friends! I don't think a lot of users will want to (or feel safe doing) edit the Windows registry just to hack this. Maybe this will make it into the next build.

**Comment by drydh** [comment]
Glad to hear that it worked out.

**Comment by HeyItsJustMe** [comment]
Do you think this could be fixed in the next release?

**Comment by HughResnick** [comment]
Running Frescobaldi on Win10. Not being very technical, I held my breath, and went into regedit. I was only able to dig down to HKEY_CURRENT_USER\SOFTWARE\frescobaldi\frescobaldi\midi where the choices I had were (Default), close_outputs, and polling_time. 

Am I doing this right, and, if so, what are my next steps? Thanks.

**Comment by 19joho66** [comment]
I hvan't got a real midi device connected, but I've tried it with an virtual midi keyboard via loopMidi Port. The picture shows where to set the input port.

![midi Setting](https://user-images.githubusercontent.com/15361970/80635175-d6e1ca00-8a5b-11ea-9bbe-77a6ab3af5d3.png)
I was then able to record midi notes. Hope this works for you

**Comment by HughResnick** [comment]
Thank you! Solved it.

**Comment by wbsoft** [comment]
Does Frescobaldi still need a fix? Or does it work correctly now? (Then this br can be closed)

**Comment by HeyItsJustMe** [comment]
The regedit hack above technically fixes the issue but it's more of a 
workaround than a true fix. The user must hack the registry in order for 
Frescobaldi to work. If you think this is ""fixed"" then sure, it's 
fixed I guess.

**Comment by HughResnick** [comment]
No. At least not due to my question. I was an idiot who was running 2.20 instead of the most recent version. As soon as I upgraded, the problem went away.

**Comment by fedelibre** [comment]
> I don't have a Windows machine but looking at other issues, the key in the registry is probably (at least under Windows 8.1) called:
> 
> HHKEY_CURRENT_USER\software\Frescobaldi\frescobaldi\midi\midi\input_port
> 
> and should be changed to
> 
> HHKEY_CURRENT_USER\software\Frescobaldi\frescobaldi\midi\input_port

I'm testing Frescobaldi 4.0.1 (Qt6). The MIDI keyboard is detected but nothing happens when I press the keys.

I cannot see that midi key. Here's what I see:

![Image](https://github.com/user-attachments/assets/6374924c-be94-4786-821f-b4aa6566b62a)

**Comment by fedelibre** [comment]
Well, the fix is really trivial.
I'll send a PR and build the .msi package so the Window users can test it.

MIDI input is working here, but it's slow. In Linux I could get it working smoothly by adjusting some settings in the MIDI synth, but I'm not confident with Windows at all.

**Comment by fedelibre** [comment]
Just switched to Linux. For the records, the setting which allowed me to input notes without any delay is *Buffer size: 128*:

![Image](https://github.com/user-attachments/assets/64015d6f-2f55-459b-936a-dac805a74979)

**Comment by fedelibre** [comment]
@HeyItsJustMe You can try installing this .msi file and see if it's working now:
https://github.com/frescobaldi/frescobaldi/actions/runs/14412580726/artifacts/2930791526

**Comment by StevestonSurfer** [comment]
Midi input appears not to be working on Windows 11 Home. It works with my other software so the issue appears to be confined to Frescobaldi. Please help! I love Frescobaldi and would love to enable this feature!

**Comment by ksnortum** [comment]
This was user error, see #2219.
